### PR TITLE
Update resample time offset FutureWarning and docs

### LINKED ISSUE #7596: illustrate time offset arithmetic
### Is your feature request related to a problem?

We should document changing the time vector using pandas date offsets [here](https://docs.xarray.dev/en/stable/user-guide/time-series.html#time-series-data)

This is particularly useful for centering the time stamps after a resampling operation.

Related: 
- CFTime offsets: https://github.com/pydata/xarray/issues/5687
- `loffset` deprecation: https://github.com/pydata/xarray/pull/7444

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by douglatornell**
Improve the text of the `FutureWarning` re: deprecation of the `loffset` parameter in the `DataWithCoords._resample()`.
Add deprecation notices about `loffset` to the `Dataset.resample()` and DataArray.resample()` docstrings.
Add illustration of updating the time coordinate values of a resampled dataset using  time offset arithmetic to the Time series data - Resampling and grouped operations section of the user guide.

- [x] Closes #7596
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

File: xarray/core/common.py [hunk 1]
```diff
@@ -1010,8 +1010,10 @@ def _resample(
 
         if loffset is not None:
             emit_user_level_warning(
-                "Following pandas, the `loffset` parameter to resample will be deprecated "
-                "in a future version of xarray.  Switch to using time offset arithmetic.",
+                "FutureWarning: Following pandas, the `loffset` parameter to resample "
```
Comment (max-sixty) [inline xarray/core/common.py:1013]:
```suggestion
                "Following pandas, the `loffset` parameter to resample "
```

**Review by max-sixty — APPROVED** [review]
Looks good! Thanks @douglatornell

File: xarray/core/common.py (hunk 1 above)
Comment (douglatornell) [reply xarray/core/common.py:1013]: DOH! :man_facepalming: Will fix.

**Comment by douglatornell** [comment]
Thanks for the lightning fast review @max-sixty !

**Comment by douglatornell** [comment]
@dcherian Does my addition to the bottom of `time-series.rst` accomplish what you had in mind in #7596?

File: doc/user-guide/time-series.rst [hunk 2]
```diff
@@ -245,6 +245,18 @@ Data that has indices outside of the given ``tolerance`` are set to ``NaN``.
...
+That can be accomplished by updating the resampled dataset time coordinate values
+using time offset arithmetic via the `pandas.tseries.frequencies.to_offset`_ function.
+
+.. _pandas.tseries.frequencies.to_offset: https://pandas.pydata.org/docs/reference/api/pandas.tseries.frequencies.to_offset.html
+
+.. ipython:: python
+
+    resampled_ds = ds.resample(time="6h").mean()
+    offset = pd.tseries.frequencies.to_offset("6h") / 2
+    resampled_ds["time"] = resampled_ds.get_index("time") + offset
```
Comment (max-sixty) [inline doc/user-guide/time-series.rst:258]:
```suggestion
    resampled_ds["time"] = resampled_ds.indexes["time"] + offset
```

Do we prefer one of these than the other? I didn't even know about `.get_index`...

File: xarray/core/common.py [hunk 3]
```diff
@@ -1010,8 +1010,10 @@ def _resample(
 
         if loffset is not None:
             emit_user_level_warning(
-                "Following pandas, the `loffset` parameter to resample will be deprecated "
-                "in a future version of xarray.  Switch to using time offset arithmetic.",
+                "Following pandas, the `loffset` parameter to resample "
+                "will be deprecated in a future version of xarray.  "
```
Comment (max-sixty) [inline xarray/core/common.py:1014]:
```suggestion
                "is deprecated in a future version of xarray.  "
```

should we take this opportunity to convert to the present tense? :)

**Comment by max-sixty** [comment]
I'll wait for @dcherian to have a look before hitting the big button

File: doc/user-guide/time-series.rst (hunk 2 above)
Comment (douglatornell) [reply doc/user-guide/time-series.rst:258]: The [v0.9.0 changelog](https://docs.xarray.dev/en/stable/whats-new.html#id1112) seems to suggest that `get_index()` is more generally robust than the `indexes` property, though I'm not sure that it matters in this instance.

File: xarray/core/common.py [hunk 4]
```diff
@@ -1010,8 +1010,10 @@ def _resample(
 
         if loffset is not None:
             emit_user_level_warning(
-                "Following pandas, the `loffset` parameter to resample will be deprecated "
-                "in a future version of xarray.  Switch to using time offset arithmetic.",
+                "Following pandas, the `loffset` parameter to resample "
+                "is deprecated in a future version of xarray.  "
```
Comment (max-sixty) [inline xarray/core/common.py:1014]:
```suggestion
                "is deprecated.  "
```

Comment (dcherian) [reply xarray/core/common.py:1014]: How about

```suggestion
                "is deprecated.  For example, 
                    >>> offset = pd.tseries.frequencies.to_offset(freq) / 2
    		    >>> resampled_ds["time"] = resampled_ds.get_index("time") + offset "
```

Seems like we should just support a `center=True`

**Review by dcherian — APPROVED** [review]

**Pull request merged**